api/include/opentelemetry/trace/propagation/http_trace_context.h

```diff
@@ -81,22 +82,25 @@ class HttpTraceContext : public HTTPTextFormat<T>
     return context.SetValue(span_key, sp);
   }
 
-  static Span *GetCurrentSpan(const context::Context &context)
+  static void GetCurrentSpan(const context::Context &context, SpanContext &span_context)
   {
     const nostd::string_view span_key = "current-span";
     context::Context ctx(context);
-    nostd::shared_ptr<Span> span = nostd::get<nostd::shared_ptr<Span>>(ctx.GetValue(span_key));
-    return (span.get());
+    context::ContextValue span = ctx.GetValue(span_key);
+    if ((int64_t)span == 0)
+    {
+      return;
+    }
+    span_context = ((Span)span).GetContext();
```

Comment:
This doesn't look right. `span` is a variant, so the int64_t alternative being 0 doesn't indicate anything about whether the shared_ptr<Span> alternative is non-null. Also, getting the wrong alternative is typically an error.

Use `std::holds_alternative` to see if it has the right type, and then check if the shared ptr is null.

Reply:
Yes I do find that problemetic, and it's causing my program to fail. I am currently trying to find a way to solve this. Do you have any suggestions?

Reply:
The GetValue returns a 0 if it is not found, that's why put if ((int64_t)span == 0) there. Would this fail?

Reply:
I think this problem is resolved now with the span context being a invalid SpanContext() when extraction from context fails.